tests/test_toa_indexing.py

```diff
@@ -57,7 +51,6 @@ def test_getitem_boolean(c):
     assert len(s) == np.sum(c)
     assert np.all(s.get_mjds() == m[c])
     if len(s) > 0:
-        assert np.all(s.table["mjd_float"] == s.table.group_by("obs")["mjd_float"])
```

Comment:
I think this test differs from the previous in that it checks that goruping is preserved, so it can go.